crates/rattler_installs_packages/src/resolve/dependency_provider.rs

```diff
@@ -49,7 +66,15 @@ impl Display for PypiVersionSet {
 /// within the [`PypiVersionSet`] version set.
 #[derive(Clone, Debug, Ord, PartialOrd, Eq, PartialEq)]
 pub(crate) enum PypiVersion {
-    Version(Version),
+    Version {
+        version: Version,
+
+        /// This is true if there are only pre-releases available for this package
+        /// For example, if the package `foo` has only versions `foo-1.0.0a1` and `foo-1.0.0a2`
+        /// then this will be true. This allows us later to match against this version and
+        /// allow the selection of pre-releases.
```

Comment:
Comment needs to be updated to account for all cases that this can be set to true :)